### ISSUE TITLE
Filtering by null seems to be broken in 2.5

### ISSUE THREAD

**Issue by jwld** [prisma/prisma#3352]
## Bug description

Encountering a couple of situations where filtering by null is now throwing an error.

## How to reproduce

**Schema**
```
model Visit {
  id        String    @default(cuid()) @id
  rating    Int?
  skippedAt DateTime?
}
```

**Error 1**
```
const res = await prisma.visit.findMany({
  where: {
    OR: [
      { rating: 3 },
      { skippedAt: null }
    ]
  }
})
```
```
PrismaClientKnownRequestError:
Invalid `prisma.visit.findMany()` invocation in
/Users/JackDavies/Desktop/prisma-issue-2/run.js:16:34

  15
→ 16 const res = await prisma.visit.findMany(
  Failed to validate the query `Error occurred during query validation & transformation:
Query (object)
  ↳ findManyVisit (field)
    ↳ where (argument)
      ↳ VisitWhereInput (object)
        ↳ OR (field)
          ↳ VisitWhereInput (object)
            ↳ skippedAt (field)
              ↳ DateTimeNullableFilter (object)
                ↳ equals (field)
                  ↳ Value types mismatch. Have: Object({"equals": Null}), want: Scalar(DateTime)` at `.Query.findManyVisit.where.VisitWhereInput.OR.VisitWhereInput.skippedAt.DateTimeNullableFilter.equals`
    at PrismaClientFetcher.request (/Users/JackDavies/Desktop/prisma-issue-2/node_modules/@prisma/client/runtime/index.js:1:226003)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async main (/Users/JackDavies/Desktop/prisma-issue-2/run.js:16:15) {
  code: 'P2009',
  meta: {
    query_validation_error: 'Error occurred during query validation & transformation:\n' +
      'Query (object)\n' +
      '  ↳ findManyVisit (field)\n' +
      '    ↳ where (argument)\n' +
      '      ↳ VisitWhereInput (object)\n' +
      '        ↳ OR (field)\n' +
      '          ↳ VisitWhereInput (object)\n' +
      '            ↳ skippedAt (field)\n' +
      '              ↳ DateTimeNullableFilter (object)\n' +
      '                ↳ equals (field)\n' +
      '                  ↳ Value types mismatch. Have: Object({"equals": Null}), want: Scalar(DateTime)',
    query_position: '.Query.findManyVisit.where.VisitWhereInput.OR.VisitWhereInput.skippedAt.DateTimeNullableFilter.equals'
  }
}
```

**Error 2**
```
const res = await prisma.visit.aggregate({
  count: true,
  where: { rating: { not: null } }
})
```

```
PrismaClientKnownRequestError:
Invalid `prisma.visit.aggregate()` invocation in
/Users/JackDavies/Desktop/prisma-issue-2/run.js:11:32

   7
   8   // await prisma.visit.create({ data: { rating: 1 } })
   9   // await prisma.visit.create({ data: { rating: 4 } })
  10
→ 11 const res = await prisma.visit.aggregate(


  Missing a required value at `Query.aggregateVisit.where.VisitWhereInput.rating.IntNullableFilter.not`
    at PrismaClientFetcher.request (/Users/JackDavies/Desktop/prisma-issue-2/node_modules/@prisma/client/runtime/index.js:1:226003)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async main (/Users/JackDavies/Desktop/prisma-issue-2/run.js:11:13) {
  code: 'P2012',
  meta: {
    path: 'Query.aggregateVisit.where.VisitWhereInput.rating.IntNullableFilter.not'
  }
}
```

## Expected behavior

Should be able to filter by null on nullable fields as before.

## Environment & setup

- OS: Mac OS
- Database: PostgreSQL
- Node.js version: 14.5.0
- Prisma version:

```
2.5.0
```

**Comment by janpio** [comment]
Possibly related: https://github.com/prisma/prisma/issues/3342

**Comment by sejr** [comment]
I'm experiencing this with a non-null filter query.

**Comment by AhmedElywa** [comment]
I have same issue

**Comment by sejr** [comment]
@AhmedElywa upgrade to the 2.6 dev version; that's what fixed it for me.

**Comment by Jolg42** [comment]
We issued 2 fixes in a dev build that could already solve your problems, if you want to try change your version to 2.5.1-dev.3 and let us know 😃

(do not use 2.6 dev version)

**Comment by jwld** [comment]
Thanks will do - any chance this will get a patch release? Keen to upgrade but stuck on 2.2 still

**Comment by janpio** [comment]
> any chance this will get a patch release?

Yes, the `dev` version of `2.5.1` will later be released as `2.5.1` on `@latest` as soon as we know it fixes the problem that were reported.

**Comment by Jolg42** [comment]
This is fixed in `2.5.1-dev.4` and will be part of the 2.5.1 official patch. Thanks for the issue 😃

**Comment by Jolg42** [comment]
You can now use [the official 2.5.1 patch](https://github.com/prisma/prisma/releases/tag/2.5.1) 🎊
